pkg/agent/controller/traceflow/packetin.go

```diff
@@ -173,6 +173,17 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*crdv1beta1.Traceflo
 		ob := new(crdv1beta1.Observation)
 		ob.Component = crdv1beta1.ComponentSpoofGuard
 		ob.Action = crdv1beta1.ActionForwarded
+		// For SNATed packet(hairpin), ipSrc and ctNwSrc are different.
+		// We noticed that ctNwSrc is invalid for ICMPv6 packets: it should contain
+		// the original src Pod IP but it is always empty due to an issue in OVS.
+		// https://github.com/openvswitch/ovs-issues/issues/327
+		if isValidCtNw(ctNwSrc) {
+			ob.SrcPodIP = ctNwSrc
+		} else {
+			// Incase of ICMPv6, since ctNwSrc is invalid, we can use ipSrc as
+			// hairpin is not applicable in ICMPv6, so ipSrc always contain src pod IP.
```

Comment:
```suggestion
			// In the case of ICMPv6, since ctNwSrc is invalid, we can use ipSrc as
			// hairpin is not applicable, so ipSrc always contains src pod IP.
```